### PR TITLE
[Windows] Fixes various tests

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>
Commit Message:

Fixes various tests on Windows

Additional Description:

For all the tests that I have removed the tag I have tested them locally on 100+ runs to make sure that they are not flaky.

Risk Level: N/A test only
Testing: test only
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
cc: @envoyproxy/windows-dev, before jumping into review give one CI run to fix any format errors

File: test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc [hunk 1]
```diff
@@ -345,6 +345,13 @@ TEST_P(ProxyProtocolTest, ErrorRecv_2) {
   EXPECT_CALL(os_sys_calls, close(_)).Times(AnyNumber()).WillRepeatedly(Invoke([this](os_fd_t fd) {
     return os_sys_calls_actual_.close(fd);
   }));
+#ifdef WIN32
```
Comment (htuch) [inline test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:348]: This doesn't seem ideal, can we avoid special casing Windows in test that aren't platform specific?

Comment (davinci26) [reply test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:348]: Sure, let me see what is the best way to go about removing the `#ifdef`

Comment (davinci26) [reply test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc]: Thanks for the review, fixed in 9319ca061d5ecdc64f3a9c135393f59d62e5138b

File: test/integration/proxy_proto_integration_test.cc [hunk 2]
```diff
@@ -137,7 +137,7 @@ TEST_P(ProxyProtoIntegrationTest, AccessLog) {
 
   ASSERT_EQ(2, tokens.size());
   EXPECT_EQ(tokens[0], Network::Test::getLoopbackAddressString(GetParam()));
-  EXPECT_EQ(tokens[1], "1.2.3.4:12345\n");
+  EXPECT_EQ(tokens[1], fmt::format("1.2.3.4:12345{}", TestEnvironment::newLine));
```
Comment (htuch) [inline test/integration/proxy_proto_integration_test.cc:140]: Trying to wrap my head around why proxy proto would behave different for Windows/Linux? Presumably new lines here are network wire features, not files.

File: test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc [hunk 3]
```diff
@@ -345,6 +345,12 @@ TEST_P(ProxyProtocolTest, ErrorRecv_2) {
   EXPECT_CALL(os_sys_calls, close(_)).Times(AnyNumber()).WillRepeatedly(Invoke([this](os_fd_t fd) {
     return os_sys_calls_actual_.close(fd);
   }));
+  EXPECT_CALL(os_sys_calls, accept(_, _, _))
+      .Times(AnyNumber())
+      .WillRepeatedly(Invoke(
```
Comment (htuch) [inline test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:350]: I think this is an unfortunate use of mocking to provide real syscall behavior, but it's consistent with the existing code which basically establishes this pattern, so I think it's OK for this PR. Maybe add some TODOs above each of these blocks to clean this up in the future, this is really bad test code.

Comment (davinci26) [reply test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:350]: Sure makes sense. Do you prefer a TODOs above the code blocks or an issue on github?

Comment (davinci26) [reply test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:350]: Also as an observation I have seen this pattern before in many tests.

File: test/integration/proxy_proto_integration_test.cc (hunk 2 above)
Comment (davinci26) [reply test/integration/proxy_proto_integration_test.cc:140]: I haven't delved much into it, initially I though it came from the way we save the access log file but I could be wrong. I will investigate a bit more and get back at you.

Comment (wrowe) [reply test/integration/proxy_proto_integration_test.cc:140]: Carriage control/line endings seem really weird in this context, I'd expect all whitespace to fall off once tokens are split.

Comment (davinci26) [reply test/integration/proxy_proto_integration_test.cc:140]: by default `split` does not trim whitespace. But that would be an alternative way to get the test working.

Comment (davinci26) [reply test/integration/proxy_proto_integration_test.cc:140]: I think that would be more robust in light of different type of line endings.

File: test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc (hunk 3 above)
Comment (davinci26) [reply test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc:355]: Added a TODO in code for this in 8590fee058d197a3888261c1a541e2fd449d2c17. Let me know if I should also open an issue.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**